### PR TITLE
Sticky sessions

### PR DESCRIPTION
**Pull request by felipemadero**
## Why this should be merged
API clients do not use sticky sessions, causing sometimes inconsistencies among different calls. Same applies to wallet.

## How this works
Sticky session is set for all API clients, by making rpc requester to remember and pass cookies to the server.
Wallet now uses the same pClient and xClient during all the wallet lifetime (previously eg one pClient was used for getting  utxos and other pClient was used to issue the tx creation calls).

## How this was tested
Tested on CLI for the commands key transfer and subnet elastic, that previously used Sleep calls to avoid inconsistencies.

### CODE REVIEW

**Comment by danlaine** [comment]
Can't we use the `options ...Option` argument in `SendJSONRequest` to attach cookies?

File: utils/rpc/options.go [hunk 1]
```diff
@@ -6,19 +6,23 @@ package rpc
...
+
+	"golang.org/x/exp/maps"
 )
 
 type Option func(*Options)
 
 type Options struct {
 	headers     http.Header
 	queryParams url.Values
+	cookies     map[string]*http.Cookie
```
Comment (danlaine) [inline utils/rpc/options.go:18]: Can this just be a `[]*http.Cookie`? Seems like we never use the keys in this map

File: utils/rpc/requester.go [hunk 2]
```diff
@@ -36,12 +39,22 @@ func (e *avalancheEndpointRequester) SendRequest(
...
 		uri,
 		method,
 		params,
 		reply,
 		options...,
 	)
+
+	for _, c := range newCookies {
+		e.cookies[c.Name] = c
+	}
```
Comment (StephenButtolph) [inline utils/rpc/requester.go:57]: I think this makes all of our clients non-thread safe... right?

**Review by StephenButtolph — COMMENTED** [review]
I'm wondering if passing cookies this way is just a big anti-pattern. I feel like we should just create an `http.Client` with a non-nil `CookieJar` if we wanted to support sticky sessions. That will also provide safer handling around which cookies are sent.

**Comment by StephenButtolph** [comment]
This doesn't seem needed anymore. If we were to decide to forward the cookies - we would change the `http.Client` to have a `CookieJar`.

**Pull request closed without merging**